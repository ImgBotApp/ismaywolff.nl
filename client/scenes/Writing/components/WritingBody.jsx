import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import dateformat from 'dateformat'
import { gutter } from '../../../styles'
import { selectors } from '../../../data/articles'
import { Cell, Grid } from '../../../components/grid'
import { InternalLink } from '../../../components/links'
import { Spinner } from '../../../components/spinner'
import { AppError } from '../../../components/errors'
import { TextLine, TextLineFill, TextLineFit } from '../../../components/text'

export function WritingBody({ entities, articles }) {
  // if fetching or hasn't fetched yet
  if (articles.isFetching || !articles.didFetch) {
    return <Spinner />
  }

  // if there's an error
  if (articles.hasError) {
    return <AppError error={articles.errorMessage} />
  }

  return (
    <Grid gutter={gutter}>
      {articles.result.map(id => (
        <Cell
          gutter={gutter}
          key={id}
        >
          <InternalLink to={`/writing/${entities[id].slug}`}>
            <TextLine>
              <TextLineFill direction={'right'}>{entities[id].title}</TextLineFill>
              <TextLineFit>{dateformat(entities[id].published, 'mmmm yyyy')}</TextLineFit>
            </TextLine>
          </InternalLink>
        </Cell>
      ))}
    </Grid>
  )
}

WritingBody.propTypes = {
  entities: PropTypes.object.isRequired,
  articles: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  entities: selectors.getArticleEntities(state),
  articles: selectors.getArticleState(state)
})

export default connect(mapStateToProps)(WritingBody)
