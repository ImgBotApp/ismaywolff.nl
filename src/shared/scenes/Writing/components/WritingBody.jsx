import React from 'react'
import { object, objectOf, shape, bool, string, arrayOf } from 'prop-types'
import { connect } from 'react-redux'
import dateformat from 'dateformat'
import { gutter } from '../../../styles'
import { selectors } from '../../../data/articles'
import { Cell, Grid } from '../../../components/grid'
import { InternalLink } from '../../../components/links'
import { Spinner } from '../../../components/spinner'
import { AppError } from '../../../components/errors'
import { TextLine, TextLineFill, TextLineFit } from '../../../components/text'

export function DumbWritingBody({ entities, articles }) {
  // If fetching or hasn't fetched yet
  if (articles.isFetching || !articles.didFetch) {
    return <Spinner />
  }

  // If there's an error
  if (articles.errorMessage) {
    return <AppError errorMessage={articles.errorMessage} />
  }

  return (
    <Grid gutter={gutter}>
      {articles.result.map(id => (
        <Cell gutter={gutter} key={id}>
          <InternalLink to={`/writing/${entities[id].slug}`}>
            <TextLine>
              <TextLineFill direction={'right'}>{entities[id].title}</TextLineFill>
              <TextLineFit>{dateformat(entities[id].published, 'mmmm yyyy')}</TextLineFit>
            </TextLine>
          </InternalLink>
        </Cell>
      ))}
    </Grid>
  )
}

DumbWritingBody.propTypes = {
  entities: objectOf(object).isRequired,
  articles: shape({
    didFetch: bool.isRequired,
    errorMessage: string.isRequired,
    isFetching: bool.isRequired,
    result: arrayOf(string).isRequired
  }).isRequired
}

const mapStateToProps = state => ({
  entities: selectors.getArticleEntities(state),
  articles: selectors.getArticleState(state)
})

export default connect(mapStateToProps)(DumbWritingBody)
