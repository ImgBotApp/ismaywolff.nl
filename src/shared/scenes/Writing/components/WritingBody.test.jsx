import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbWritingBody } from './WritingBody'

describe('<WritingBody />', () => {
  it('renders correctly', () => {
    const entities = {
      one: {
        slug: 'slug',
        published: '10-12-2000',
        title: 'title'
      }
    }
    const articles = {
      result: ['one'],
      didFetch: true,
      errorMessage: '',
      isFetching: false
    }
    const wrapper = shallow(<DumbWritingBody entities={entities} articles={articles} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders a loading state', () => {
    const articles = {
      result: [],
      didFetch: false,
      errorMessage: '',
      isFetching: true
    }
    const wrapper = shallow(<DumbWritingBody entities={{}} articles={articles} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders errors', () => {
    const articles = {
      result: [],
      didFetch: true,
      errorMessage: 'Something went wrong',
      isFetching: false
    }
    const wrapper = shallow(<DumbWritingBody entities={{}} articles={articles} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
