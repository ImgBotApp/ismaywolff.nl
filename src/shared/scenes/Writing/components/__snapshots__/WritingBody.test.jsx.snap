// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WritingBody /> renders a loading state 1`] = `<styled.div />`;

exports[`<WritingBody /> renders correctly 1`] = `
<div>
  <styled.div
    bottom="2rem"
  >
    <WritingItem
      article={
        Object {
          "published": "10-12-2000",
          "slug": "slug",
          "summary": "summary",
          "title": "title",
        }
      }
    />
  </styled.div>
</div>
`;

exports[`<WritingBody /> renders errors 1`] = `
<AppError
  errorMessage="Something went wrong"
/>
`;
